websocket/connection.go

```diff
@@ -92,6 +92,25 @@ func NewDurableSendingConnection(target string, logger *zap.SugaredLogger) *Mana
 	return NewDurableConnection(target, nil, logger)
 }
 
+// NewDurableSendingConnectionGuaranteed creates a new websocket connection
+// that can only send messages to the endpoint it connects to. It returns
+// the connection if the connection can be established within the given
+// `duration`. Otherwise it returns nil.
+//
+// The connection will continuously be kept alive and reconnected
+// in case of a loss of connectivity.
+func NewDurableSendingConnectionGuaranteed(target string, duration time.Duration, logger *zap.SugaredLogger) *ManagedConnection {
```

Comment:
Should we return an error if we don't create the connection in the desired timeframe? Just returnin `nil` feels like abusing the type system a little imo. We could potentially use the error to surface the reason too.

Reply:
done